src/main/java/com/gildedrose/AgeableController.java

```diff
@@ -0,0 +1,48 @@
+package com.gildedrose;
+
+import com.gildedrose.items.Ageable;
+import waiter.Request;
+import waiter.Response;
+import waiter.ResponseBuilder;
+
+import java.util.function.Function;
+
+public class AgeableController {
```

Comment:
Question: I'm a little surprised to see an `Ageable` controller rather than an `Item` controller. The same goes for the repository naming convention.

Could you discuss why this approach was taken to create a repository and with the interface in mind rather than the objects/resources that adhere to that interface?

Suggestion: Are there any tests associated with this new controller? I'm also a little surprised to see JSON formatting being done in this controller file, perhaps this could be extracted and unit tested?

Reply:
+1 to what Eli said above

Reply:
I'm making an educated guess as to why the name `AgeableController` was chosen. Let me know if my assumption is off base.

In some cases we prefer to use the abstraction on the left-hand side of assignment to give us flexibility to swap out implementations like this:

```
// Concrete types don't give us room to change without breaking.
ArrayList myList = new ArrayList();

// Using the abstract type on the left hand side means I can swap out ArrayList for LinkedList without the client code, and the compiler having to care.
List myList = new ArrayList();
```

However in this case, there is no other Controller object that we're going to be swapping out for this one. Therefore we can use more concrete type names like `ItemController` without causing issues like what I described earlier.

This is similar to how Controllers are named in frameworks like Rails (e.g. `ItemController < ActionController`) and Spring Boot (https://www.baeldung.com/spring-boot-start#simple-mvc-view).

Reply:
That was indeed the thinking, but after thinking it over and talking about it with Ish, Steven and Eli, we came to the conclusion that it would make more sense for the Controller, Repository and Presenter all to follow the `Items` prefix naming convention!

The Repository naming convention is displayed in [this commit](https://github.com/jramnani/GildedRoseJavaLLC/pull/24/commits/a9f211b3c0f4cb9bdc4dabaee9b2dbe283740147), and the others naming conventions can be seen in [this commit](https://github.com/jramnani/GildedRoseJavaLLC/pull/24/commits/47f6050bffa52b45ae2b47515696b6c15fe88a33) (and onward)!